lib/hooks.ts

```diff
@@ -30,13 +30,17 @@ export function useProducts() {
     };
 }
 
-export function useProductList() {
+export function useProductList(page?: number, limit?: number) {
     const encodedContext = useSession()?.context;
+    const pageQuery = page ? `&page=${page}` : '';
+    const limitQuery = limit ? `&limit=${limit}` : '';
+    const query = `${pageQuery}${limitQuery}`;
```

Comment:
this looks like it could get a bit messy, especially if we need to add additional params.  to make this a bit more dynamic, we can make use of `URLSearchParams`. **Note**: I recommend changing the props to a props Object:

e.g.
```jsx
// Place in the ../types folder somewhere
interface QueryProps {
    [key: string]: string | number | boolean; // might be able to expand QueryParams interface
}

export function useProductList(query?: QueryProps) {
    const encodedContext = useSession()?.context;

    const queryEntries = Object.entries(query);
    const params = new URLSearchParams();
    if (queryEntries.length) {
        queryEntries.forEach(([key, value]) => params.append(key, String(value)));
    }
    // Use an array to send multiple arguments to fetcher
    const { data, error, mutate: mutateList } = useSWR(encodedContext ? ['/api/products/list', encodedContext, params.toString()] : null, fetcher);
    ...
}
```
And then you'll call the hook like `useProductList({ page: 2, limit: 10 })`

Reply:
now that I'm looking at this file, I'm realizing that we're passing `encodedContext` as a param - this might actually be a good time to just make the other hooks more URL param friendly. We can actually modify the sample code I gave above such that we send only one param (query string) to the fetcher function:
```jsx
async function fetcher(url: string, query: string) {
    const res = await fetch(`${url}?${query}`);
    ...
}

export function useProducts() {
    const encodedContext = useSession()?.context;
    const params = new URLSearchParams();
    params.append('context', encodedContext);
    // Request is deduped and cached; Can be shared across components
    const { data, error } = useSWR(encodedContext ? ['/api/products', params.toString()] : null, fetcher);

    return {
        summary: data,
        isLoading: !data && !error,
        error,
    };
}

export function useProductList(query?: QueryProps) {
    const encodedContext = useSession()?.context;
    const params = new URLSearchParams();
    params.append('context', encodedContext);

    const queryEntries = Object.entries(query);
    if (queryEntries.length) {
        queryEntries.forEach(([key, value]) => params.append(key, String(value)));
    }
    // Use an array to send multiple arguments to fetcher
    const { data, error, mutate: mutateList } = useSWR(encodedContext ? ['/api/products/list', params.toString()] : null, fetcher);
    ...
}
```